enhancements/baremetal/baremetal-config-raid-and-bios.md

```diff
@@ -0,0 +1,149 @@
+has supported the process of RAID and BIOS for worker nodes).
+
+### User Stories
+
+With the addition of this feature, the users can configure RAID and BIOS in IPI deployments.
+
+### Implementation Details/Notes/Constraints
+
+#### Fill two fields
+
+The user runs `create manifests` and then fill *raid* and *firmware* fields in the **BMH** created by the installer.
+
+```yaml
+spec:
+  bmc:
+    address: <bmc>://<ip-address>
+    credentialsName: worker-0-bmc-secret
+  bootMACAddress: mac-address
+  raid: raid-config
+  firmware: bios-config
+  ...
+```
+
+#### Process the fields in installer
+
+For control plane hosts, transform the *raid* and *firmware* fields into string respectively and then write them into the
+***terraform.baremetal.auto.tfvars.json*** file as the values of *raid_config* and *bios_settings* respectively.
+
+#### Process the fields in terraform-provider-ironic
+
+Add *raid_config* and *bios_settings* fields to terraform-provider-ironic API, transform them back to struct format, and then call the
+methods in **BMO** to process the two fields to perform [manual cleaning](https://docs.openstack.org/ironic/latest/admin/cleaning.html#manual-cleaning).
+
+#### Notes
+
+When the IPI deployments completed, the **BMH** resources will still get persisted in the cluster, though control plane hosts have
+RAID and BIOS field in their **BMH**, but the *externallyProvisioned* field's value is true, so the RAID and BIOS configurations are
+invalid.
+
+### Risks and Mitigations
+
+N/A
+
+## Design Details
+
+### Test Plan
+
+- Unit tests for determining the configuration of RAID and BIOS passed to Ironic meeting expectations.
+- e2e tests for determining the configuration of RAID and BIOS configured during the IPI deployments.
+
+### Graduation Criteria
+
+N/A
+
+#### Dev Preview -> Tech Preview
+
+N/A
+
+#### Tech Preview -> GA
+
+N/A
+
+#### Removing a deprecated feature
+
+N/A
+
+### Upgrade / Downgrade Strategy
+
+Older versions of the installer will ignore the *raid* and *firmware* fields.
+
+### Version Skew Strategy
+
+N/A
+
+## Implementation History
+
+- [Implementation on openshift/installer](https://github.com/hs0210/installer/tree/ipi-support-raid-bios)
+- [Implementation on openshift-metal3/terraform-provider-ironic](https://github.com/hs0210/terraform-provider-ironic/tree/ipi-support-raid-bios)
+
+## Drawbacks
+
+This will increase the number of steps in IPI deployments and take longer.
+
+## Alternatives
+
+Add RAID and BIOS configuration by modifying the manifest files **~/clusterconfigs/openshift/99_openshift-cluster-api_hosts- \*.yaml** generated by executing `openshift-baremetal-install --dir ~/clusterconfigs create manifest`.
+
+This approach takes advantage of already supported features, so it does not need to modify any source code, but it only works for worker nodes.
```

Comment:
Another alternative is to add this data to the install-config, but we discussed it and decided modifying manifests would be a reasonable alternative, given that this is an "advanced" API, and that it's not yet supported for all vendors/drivers.

Reply:
Yes, I have updated the Alternatives section.